src/NetAnalyzers/VisualBasic/Microsoft.NetCore.Analyzers/Runtime/BasicSpecifyCultureForToLowerAndToUpper.Fixer.vb

```diff
@@ -21,25 +21,41 @@ Namespace Microsoft.NetCore.VisualBasic.Analyzers.Runtime
         End Function
 
         Protected Overrides Async Function SpecifyCurrentCultureAsync(document As Document, generator As SyntaxGenerator, root As SyntaxNode, node As SyntaxNode, cancellationToken As CancellationToken) As Task(Of Document)
-            If node.IsKind(SyntaxKind.IdentifierName) Then
-                Dim invocation = node.Parent?.FirstAncestorOrSelf(Of InvocationExpressionSyntax)
-                If invocation IsNot Nothing Then
-                    Dim model = Await document.GetSemanticModelAsync(cancellationToken).ConfigureAwait(False)
-                    Dim symbolInfo = model.GetSymbolInfo(node, cancellationToken).Symbol
-                    Dim methodSymbol = TryCast(symbolInfo, IMethodSymbol)
-
-                    If methodSymbol IsNot Nothing And methodSymbol.Parameters.Length = 0 Then
-                        Dim newArg = generator.Argument(CreateCurrentCultureMemberAccess(generator, model)).WithAdditionalAnnotations(Formatter.Annotation)
-                        Dim newInvocation = invocation.AddArgumentListArguments(DirectCast(newArg, ArgumentSyntax)).WithAdditionalAnnotations(Formatter.Annotation)
-                        Dim newRoot = root.ReplaceNode(invocation, newInvocation)
-                        Return document.WithSyntaxRoot(newRoot)
-                    End If
+            If ShouldFix(node) Then
```

Comment:
Is this check needed when the parent already checking this before adding these fixes?
https://github.com/dotnet/roslyn-analyzers/blob/04edea1ac503805a8d9a6959ec18e20bff506bb5/src/NetAnalyzers/Core/Microsoft.NetCore.Analyzers/Runtime/SpecifyCultureForToLowerAndToUpper.Fixer.cs#L24

Reply:
@buyaa-n as far as I've seen in the implementation of other fixers, they mirror some of the checks made in the analyzer. The types here don't guarantee that nodes are exactly the type you expect so defensive coding is a good idea I think.

Reply:
I don't mind defensive code, but it looks to me that this method is just called twice, though I might not seeing all scenarios